aplib.net-demo/Assets/Scripts/Entities/Player/MouseLock.cs

```diff
@@ -14,25 +19,45 @@ private void Awake()
     }
 
     /// <summary>
-    /// Shows the mouse and unlocks it.
+    /// Shows the mouse and unlocks it. Usefull on websites.
     /// </summary>
     public void OnShowMousePressed()
     {
-        Cursor.lockState = CursorLockMode.None;
-        Cursor.visible = true;
-        _showMouse = true;
+        EnableMouseCurser();
     }
 
     /// <summary>
     /// On left mouse click, go back into the game and lock the cursor.
     /// </summary>
     public void OnLeftMousePressed()
     {
-        if (_showMouse)
+        if (CanvasManager.IsCursurNeeded) // if curser is needed, enable curser
         {
-            Cursor.lockState = CursorLockMode.Locked;
-            Cursor.visible = false;
-            _showMouse = false;
+            EnableMouseCurser();
         }
+        else if (_showMouse && !CanvasManager.IsCursurNeeded) // if curser is showing and is not needed, disable
+        {
+            DisableMouseCurser();
+        }
+    }
+
+    /// <summary>
+    /// Enables curser.
+    /// </summary>
+    void EnableMouseCurser()
+    {
+        Cursor.lockState = CursorLockMode.None;
+        Cursor.visible = true;
+        _showMouse = true;
+    }
+
+    /// <summary>
+    /// Disables curser.
+    /// </summary>
+    void DisableMouseCurser()
```

Comment:
Be sure to change in other locations as well
```suggestion
    void DisableMouseCursor()
```